### DISCUSSION TITLE
Error 404 with the movies and tv shows page's

### DISCUSSION THREAD

**Discussion by GabiMuresan22** [community/community#70604] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

https://gabimuresan22.github.io/Flixx-Movie-Website/ 

Here is my repository. The website is not displaying the highlight functions,  movies from the API, and also I have a problem with the TV Shows page.
I don't understood why, on my laptop everything is fine

**Comment by seyLu** [comment, marked as answer]
#### Hi @GabiMuresan22,

I've [deployed a copy of your site](https://seylu.github.io/GabiMuresan22-Flixx-Movie-Website/). Things to note:

- Add a base href on every html file:

```diff
    <title>Flixx</title>
+   <base href="/Flixx-Movie-Website/" />
  </head>    
```

- Update all links to be relative to the current page. Examples:

```diff
-  <div class="logo"><a href="/">FLIXX</a></div>
+  <div class="logo"><a href="#">FLIXX</a></div>

...

-  <a class="nav-link" href="/shows.html">TV Shows</a>
+  <a class="nav-link" href="shows.html">TV Shows</a>

...

-  <form action="search.html" class="/search-form">
+  <form action="search.html" class="search-form">

...

-  <img src="../images/no-image.jpg" class="card-img-top" alt="" />
+  <img src="images/no-image.jpg" class="card-img-top" alt="" />
```

- Since we added base href on our html file, `window.location.pathname` will return the base href + path. So we have to update that part to remove base href:

```diff
  const global = {
-   currentPage: window.location.pathname,
+   currentPage: window.location.pathname.replace(
+     "/Flixx-Movie-Website/",
+     "/",
+   ),
  };
```

PS: If you don't want to deal with manually fixing broken links when deploying to Github Pages, I'd suggest using a Static Site builder like [Astro](https://astro.build/), [Gatsby](https://www.gatsbyjs.com/), etc. or a js library/framework like [React](https://react.dev/), [Vue](https://vuejs.org/), etc. These options, they have a build step which takes care of resolving routes and links. And you can then use an NPM package, [gh-pages](https://www.npmjs.com/package/gh-pages#command-line-utility), to take care of building and deploying to github pages.

**Comment by jamesjohni83** [comment]
Hi...I was facing a similar issue on a test project too, sometimes the API loads fine locally but breaks on GitHub Pages. Might be a path or CORS issue. I’ve recently been exploring [PPCine](https://myppcine.com.br/) for streaming setups, and they handle content rendering quite smoothly, maybe check how they structure dynamic content calls. Hope this helps.
